Test changes for design picker without siteSlug [WIP]

agrullon95 opened the pull request:
#### Proposed Changes

* Changes to make Stepper Design Picker work without a siteSlug query param.

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Checkout PR
* Navigate to `/setup/free/designSetup`
* The designSetup page should still render without a `siteSlug` query parameter.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

agrullon95 committed:
Test changes for design picker without siteSlug

agrullon95 commented on client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:
```diff
@@ -530,10 +531,17 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 		);
 	}
 
```
This can be ignored for the changes required. We can pass in a prop to update the SkipLabelText value.

agrullon95 commented on client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:
```diff
@@ -409,7 +409,7 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 	function pickDesign( _selectedDesign: Design | undefined = selectedDesign ) {
 		setSelectedDesign( _selectedDesign );
-		if ( siteSlugOrId && _selectedDesign ) {
+		if ( ( isFreeFlow( flow ) || siteSlugOrId ) && _selectedDesign ) {
```
Instead of using isFreeFlow(), we can pass in a prop like, useWithoutSite, to make this component work without any siteSlug/siteId.

alshakero replied:
Yes I agree. Steps shouldn't be aware of the concept of "flow" when possible. This makes them reusable.

alshakero commented on client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:
```diff
@@ -418,12 +418,13 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
-					verticalId: siteVerticalId,
-				} ).then( () => reduxDispatch( requestActiveTheme( site?.ID || -1 ) ) )
-			);
+			siteSlugOrId &&
```
Let's make this an `if` to highlight the intentionality.

agrullon95 replied:
I removed `isFreeFlow(flow)` and added a conditional for `siteSlugOrId` to execute the `setPendingAction` logic.

alshakero commented on client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:
```diff
@@ -530,10 +531,17 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 
 	// Don't render until we've done fetching all the data needed for initial render.
-	if ( ! site || isLoadingSiteVertical || isLoadingDesigns ) {
+	if ( ( ! site && ! isFreeFlow( flow ) ) || isLoadingSiteVertical || isLoadingDesigns ) {
```
+1

agrullon95 replied:
I removed the check for `site` and `isFreeFlow(flow)` here. If site doesn't exist, it will now render.

alshakero reviewed (commented):
All in all this is great, left some comments.

agrullon95 committed:
Remove free flow conditional check

agrullon95 commented:
@alshakero - I updated the logic and added conditionals around the `siteSlugOrId` to still execute any logic related to it. This should render the design picker without a siteSlug param and will still also populate the `selectedDesign` value in the `ONBOARD_STORE`.

agrullon95 committed:
Hide paid features

agrullon95 commented on client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:
```diff
@@ -121,8 +121,18 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
-	const generatedDesigns = allDesigns?.generated?.designs || [];
-	const staticDesigns = allDesigns?.static?.designs || [];
+	// Only display free designs if siteSlugOrId does not exist
+	function filterFreeDesigns( designs: Design[] | undefined ) {
```
Added logic to hide premium designs/features since we don't want to lead users to checkout when the siteSlug query param does not exist.

Addison-Stavlo reviewed (commented):
Im just updating here for visibility:

> Added logic to hide premium designs/features since we don't want to lead users to checkout when the siteSlug query param does not exist.

We discussed this in slack/zoom and don't want to hide the premium features in the design picker. So it sounds like this PR will be updated to reflect this? (Also it sounded like which side of site creation this step exists on might change)?

Happy to review more when ready!

agrullon95 commented:
> We discussed this in slack/zoom and don't want to hide the premium features in the design picker. So it sounds like this PR will be updated to reflect this? (Also it sounded like which side of site creation this step exists on might change)?

@Addison-Stavlo - I changed this back to draft status. I will need to either update or close this PR after we confirm when we want the site creation process to occur in the free flow. I will close this PR if we go with the path of creating the site immediately after the free flow site setup screen then navigating the user to the existing design picker with their newly created site slug param in the URL.

Addison-Stavlo commented:
>  I will close this PR if we go with the path of creating the site immediately after the free flow site setup screen then navigating the user to the existing design picker with their newly created site slug param in the URL.

Sounds good! It seems that is the direction we should go for now to keep things simple and get this flow live to users, but i will let you confirm and close this.
